http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/KubernetesServiceInterp.scala

```diff
@@ -535,15 +536,20 @@ final class LeoKubernetesServiceInterp[F[_]: Parallel](
             NamespaceId(-1),
             namespaceName
           ),
-          disk,
+          Some(disk),
           req.appType match {
             case Galaxy =>
               galaxyConfig.services.map(config => KubernetesService(ServiceId(-1), config))
+            case Custom =>
+              // Back Leo will populate services when it parses the descriptor
```

Comment:
Interesting. I wonder if it would be good for front Leo to parse & validate the descriptor up front (e.g. before creating k8s resources). I can see pros and cons either way.

Reply:
Yeah, it seems simpler to bundle all the logic in back-leo, since it needs to use various information about the parsed service in the helm commands, and otherwise we'd have to duplicatively persist a lot of the descriptor information.

There's definitely value in parsing it up front, in that the user would see right away if their descriptor is invalid. I'll do a bit of testing and make sure a good message is bubbled up to the user in the case of a non-existent/unparsable descriptor.

Reply:
hmm.. this does bring up another concern that we'll have to address at some point: Versioning of these app.yamls. I expect that will be captured with the descriptor, but ill make sure to not it in the CI ticket I'll add

Reply:
Here is what the errorMessage will look like. 

```"errorMessage": "Failed to process descriptor: https://rawow.githubusercontent.com/DataBiosphere/terra-app/main/apps/ucsc_genome_browser/app.yaml. Please ensure it is a valid descriptor, and that the remote file is valid yaml following the schema detailed here: https://github.com/DataBiosphere/terra-app#app-schema. \n\tOriginal message: Error occurred fetching app descriptor from path https://rawow.githubusercontent.com/DataBiosphere/terra-app/main/apps/ucsc_genome_browser/app.yaml: \n<html>\n<head>\n<title>Fastly error: unknown domain rawow.githubusercontent.com</title>\n</head>\n<body>\n<p>Fastly error: unknown domain: rawow.githubusercontent.com. Please check that this domain has been added to a service.</p>\n<p>Details: cache-pwk4966-PWK</p></body></html>",```

Reply:
Yeah I think it's slightly better for the API to return a `Bad Request` up front than transition the app from `Creating` -> `Error` (creating k8s resources along the way) with the above error message. OK with making a separate ticket for that though.

Re versioning, the app descriptors are already stored in GitHub. And the ones in `terra-app` are really just meant to be examples, I imagine users storing them in other places as well. So Leo isn't really managing the app descriptors yet so we probably don't need to worry about versioning. (We may once we have an app "library" but that's a ways down the road.)

Reply:
Yeah, I plan to make a ticket for validation. I know we can theoretically take any app descriptor and run with it, but there is the product angle that I think should be discussed in terms of 'approved' descriptors, and whether leo should enforce that

Reply:
Though, worth mentioning, we get basic URI validation for free in front leo by nature of the decoder for the `Uri` class based on Qi's feedback (see JsonCodec)